flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/HttpRequestHandler.java

```diff
@@ -99,7 +110,43 @@ public void channelRead0(ChannelHandlerContext ctx, HttpObject msg) {
 					currentDecoder.destroy();
 					currentDecoder = null;
 				}
-				
+
+				if(secureCookie != null) {
```

Comment:
The YARN code does the auto-populaton only if security is enabled, so it should not be an issue.

> user-provided but will be persisted in the in-memory Flink configuration instance which is passed to the web layer during bootstrap.

The Configuration object in the CliFrontend is not transferred to the ApplicationMaster / JobMaster.
The flink-config.yaml (and on YARN the "dynamic properties") are the only sources of the configuration for the AM/JM.